pex/resolver.py

```diff
@@ -1382,3 +1392,83 @@ def install(
             ignore_errors=ignore_errors, max_parallel_jobs=max_parallel_jobs
         )
     )
+
+
+def resolve_from_pex(
+    pex,  # type: str
+    requirements=None,  # type: Optional[Iterable[str]]
+    requirement_files=None,  # type: Optional[Iterable[str]]
+    constraint_files=None,  # type: Optional[Iterable[str]]
+    network_configuration=None,  # type: Optional[NetworkConfiguration]
+    transitive=True,  # type: bool
+    interpreters=None,  # type: Optional[Iterable[PythonInterpreter]]
+    platforms=None,  # type: Optional[Iterable[Union[str, Platform]]]
+    manylinux=None,  # type: Optional[str]
+    ignore_errors=False,  # type: bool
+):
+    # type: (...) -> List[ResolvedDistribution]
+
+    direct_requirements = _parse_reqs(requirements, requirement_files, network_configuration)
+    direct_requirements_by_key = OrderedDict()
+    for direct_requirement in direct_requirements:
+        if isinstance(direct_requirement, LocalProjectRequirement):
+            raise Untranslatable(
+                "Cannot resolve local project {path} from the PEX repository {pex}.".format(
+                    path=direct_requirement.path, pex=pex
+                )
+            )
```

Comment:
This is because the entire category of resolves is not valid?

Is the "so what?" here to either a) remove the local project from the input requirements or b) stop using `--pex-respository`? It would be helpful to add that, if so.

Reply:
Correct - the entire category is invalid. That is the so what, but what possible other so what could it be? In other words, is there value in adding a so what when there is no alternative we can suggest besides the obvious ~ "If it hurts don't do it.". I really dislike the precedent of so much hand-holding. I totally agree that providing an alternative when there is one is something we should always do. I think explicating "False" means "No" - which is what this would be tantamount to - is just going to far.

Reply:
Hmm, on a re-read the error is unclear, should be:
```
Cannot resolve local projects from PEX repositories.  Asked to resolve {path} from {pex}.
```
At that point I stand behind the opinion above that the list of alternatives is undue handholding.

Reply:
Ok, thanks for making me look at the words more. I think this is an improvement that says everything the user needs to know: https://github.com/pantsbuild/pex/pull/1184